ReactAndroid/src/main/java/com/facebook/react/ReactInstanceManagerImpl.java

```diff
@@ -342,8 +342,12 @@ public void onPackagerStatusFetched(final boolean packagerIsRunning) {
                       public void run() {
                         if (packagerIsRunning) {
                           mDevSupportManager.handleReloadJS();
-                        } else {
+                        } else if (mDevSupportManager.hasUpToDateJSBundleInCache()) {
```

Comment:
I don't think this is correct. `recreateReactContextInBackgroundFromBundleFile` uses mJSBundleFile (e.g. the packaged JS bundle in assets). Also we check this (and whether mJSBundleFile is null) above around line 331.